File: frontend/types/scan.ts
export type ScanRun = {
  id: number;
  project_id: number;
  status: "pending" | "processing" | "completed" | "failed";
  uploaded_zip_path?: string | null;
  extracted_path?: string | null;
  total_files: number;
  supported_files: number;
  chunk_count: number;
  issue_count: number;
  started_at?: string | null;
  completed_at?: string | null;
  error_message?: string | null;
  created_at: string;
};

File: backend/app/schemas/scan.py
from pydantic import BaseModel
from datetime import datetime

class ScanRunResponse(BaseModel):
    scan_runs_id: int
    project_id: int
    status: str
    uploaded_zip_path: str | None = None
    extracted_path: str | None = None
    total_files: int = 0
    supported_files: int = 0
    chunk_count: int = 0
    issue_count: int = 0
    started_at: datetime | None = None
    completed_at: datetime | None = None
    error_message: str | None = None
    created_at: datetime

    class Config:
        from_attributes = True